src/parser/sparqlParser/SparqlQleverVisitor.h

```diff
@@ -249,12 +249,42 @@ class SparqlQleverVisitor : public SparqlAutomaticVisitor {
 
   antlrcpp::Any visitOrderClause(
       SparqlAutomaticParser::OrderClauseContext* ctx) override {
-    return visitChildren(ctx);
+    vector<OrderKey> _orderBy;
+    for (auto* orderCondition : ctx->orderCondition()) {
+      _orderBy.push_back(visitOrderCondition(orderCondition).as<OrderKey>());
+    }
+    return _orderBy;
   }
 
   antlrcpp::Any visitOrderCondition(
       SparqlAutomaticParser::OrderConditionContext* ctx) override {
-    return visitChildren(ctx);
+    if (ctx->var()) {
+      return OrderKey{VariableOrderKey(ctx->var()->getText())};
+    } else if (ctx->constraint()) {
+      auto expr = sparqlExpression::SparqlExpressionPimpl{
+          std::move(visitConstraint(ctx->constraint()).as<ExpressionPtr>())};
+      bool desc = false;
+      bool exprIsVariable = expr.getVariableOrNullopt().has_value();
+      if (exprIsVariable) {
+        string var = expr.getVariableOrNullopt().value();
+        return OrderKey{VariableOrderKey(var, desc)};
+      } else {
```

Comment:
Better scoping and less lines without loss of readability:
````
if (auto varOrNull = expr.getVariableOrNullopt(); varOrNull.has_value()) {
  return OrderKey{VariableOrder{varOrNull.value(), isDescendint}};
} else {...}
```
In the first line the separation into "initializer" and "condition" (C++17 feature) is not necessary, because the optional implicitly converts to bool, but it is still nice here to make it more visible that there is an optional involved here.